File: src/main/java/com/chatsummary/bot/repository/ChatMessageRepository.java
package com.chatsummary.bot.repository;

import com.chatsummary.bot.model.ChatMessage;
import java.time.Instant;
import java.util.List;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface ChatMessageRepository extends MongoRepository<ChatMessage, String> {

    List<ChatMessage> findByChatIdAndTimestampAfter(long chatId, Instant since);

    List<ChatMessage> findByChatIdAndTimestampBefore(long chatId, Instant before);

    void deleteByChatIdAndTimestampBefore(long chatId, Instant before);

    @Query(value = "{}", fields = "{ 'chatId': 1 }")
    List<ChatIdOnly> findAllChatIds();
}

File: src/main/kotlin/com/chatsummary/bot/repository/ChatMessageRepository.kt
package com.chatsummary.bot.repository

import com.chatsummary.bot.model.ChatMessage
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository
import java.time.Instant

@Repository
interface ChatMessageRepository : MongoRepository<ChatMessage, String> {

    fun findByChatIdAndTimestampAfter(chatId: Long, since: Instant): List<ChatMessage>

    fun deleteByChatIdAndTimestampBefore(chatId: Long, before: Instant)

    @org.springframework.data.mongodb.repository.Query(value = "{}", fields = "{ 'chatId': 1 }")
    fun findAllDistinctChatIds(): List<ChatMessage>
}
